src/vos/vos_obj_cache.c

```diff
@@ -222,6 +288,11 @@ vos_obj_hold(struct daos_lru_cache *occ, struct vos_container *cont,
 
 	create = flags & VOS_OBJ_CREATE;
 	visible_only = flags & VOS_OBJ_VISIBLE;
+	/** Pass NULL as the create_args if we are not creating the object so we avoid
+	 *  evicting an entry until we need to
+	 */
+	if (create)
+		create_flag = cont;
```

Comment:
so for 'create' the object will always be inserted into cache at the very beginning no matter if this vos_obj_hold() will success? I'm wondering why can't we treat 'create' same as 'non-create'?

Reply:
We can.  But in my benchmarking of a single object with multiple keys, it was a little faster this way.  I think probably because the create is likely to succeed.